src/Conf.ml

```diff
@@ -856,16 +857,31 @@ let action =
       (List.map !inputs ~f:(fun file ->
            { kind= kind_of file
            ; name= file
-           ; file
+           ; file= (`Input, file)
            ; conf= read_config ~filename:file config } ))
   else
     match !inputs with
+    | [] ->
+        let name = Option.value_exn !name in
```

Comment:
In general I'd prefer to document the impossible case with something like `impossible "checked by validate"` as below.